### ISSUE TITLE
Python, Java/Kotlin and Swift wrapper

### ISSUE THREAD

**Issue by DonaldTsang** [divan/txqr#4]
Adding this to https://github.com/hydrusnetwork/hydrus and associated mobile apps would be useful.  
Hydrus is written in Python, BooruNav, AnimeBoxes and Nori are on Android, Mignori is on iOS

**Comment by fionafibration** [comment]
I would agree, I'd definitely like a Python and Java wrapper. I attempted to generate a Python one using https://github.com/go-python/gopy, but I'm fresh new to Go and don't understand any of it. @divan A python binding would be appreciated, and I think I know how to do a Java one.

**Comment by DonaldTsang** [comment]
@ThePlasmaRailgun I would say the minimum would be Swift for iOS, Java/Kotlin for android and Python/JS for prototyping and scripting.

**Comment by fionafibration** [comment]
Agreed. IDK whether there are facilities for compiling/importing/binding Go into JS, Java, or Swift, but I do know Python can import compiled Go with the tool I sent above.

**Comment by NovaLogicDev** [comment]
the closest concept I have found is [gomobile](https://github.com/golang/go/wiki/Mobile), but they note the bindings have performance overhead

**Comment by fionafibration** [comment]
GoMobile should be easy bindings, but my attempts at Python bindings have failed miserably. If anyone knows more about this or has more experience, then I'd love if you could take a crack at it.

**Comment by DonaldTsang** [comment]
@ThePlasmaRailgun any progress on the TxQR java app?

**Comment by fionafibration** [comment]
Not yet, I'm still working on it.

**Comment by DonaldTsang** [comment]
@ThePlasmaRailgun is Kotlin better than Java?

**Comment by fionafibration** [comment]
We're writing it in Kotlin, I prefer it to Java.

**Comment by NovaLogicDev** [comment]
As someone who learned Java and then switched over, I can definitely say its a better language

**Comment by DonaldTsang** [comment]
Referecne: https://github.com/divan/txqr-reader/issues/2
